deploy.py

```diff
@@ -392,19 +392,17 @@ def create_tab_file(self, binaries):
     assert self.args.application
     info("Generating Tock TAB file for application/example {}".format(
         self.args.application))
-    package_parameter = "-n"
     elf2tab_ver = self.checked_command_output(["elf2tab", "--version"]).split(
-        " ", maxsplit=1)[1]
-    # Starting from v0.5.0-dev the parameter changed.
-    # Current pyblished crate is 0.4.0 but we don't want developers
-    # running the HEAD from github to be stuck
-    if "0.5.0-dev" in elf2tab_ver:
-      package_parameter = "--package-name"
+        "\n", maxsplit=1)[0]
+    if elf2tab_ver != "elf2tab 0.5.0":
```

Comment:
I would either:
* remove version check
* issue a fatal error if version is less than 0.5.0
* issue a warning if version is different/above 0.5.0

I don't see a reason to block people using 0.6.0-dev from Github.

Reply:
My concern is that doing the comparison (less than / greater than) of versions would be a much more complex logic.

I could replace the `fatal` with an `error` message, or add a `--allow-unknown-elf2tab-version` parameter to the `deploy.py` script which turns this `fatal` error into a `warning`. WDYT?

Reply:
Also, using `cargo install --vers 0.5.0 elf2tab` would guarantee that the cargo-installed elf2tab will have the tested version (https://doc.rust-lang.org/cargo/commands/cargo-install.html), so we shouldn't need to support any other version.

Another alternative would be to pin the `elf2tab` repository as a submodule in `third_party`, and build it locally instead of relying on an external binary from Cargo or from a custom directory in the user's $PATH.

Reply:
Ok then let's modify `setup.sh` to install the correct version and either remove the version check or only output a warning.